### PR TITLE
Added Posts Functionalities

### PR DESCRIPTION
**Pull request by aniketxparihar**
Added Functionalities
- Posts
- Home Feed
- Explore Feed
- Filter Feed
- Comments
- Add profile and Cover Picture to Bio

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -1 +1,2 @@
-REACT_APP_JWT_SECRET=FITME
\ No newline at end of file
+REACT_APP_JWT_SECRET=FITME
```
Comment (shraddha-amber) [inline .env:1]: never push .env on github

File: src/Components/BottomBar/BottomBar.jsx [hunk 2]
```diff
@@ -0,0 +1,110 @@
+import React from 'react'
+import "./BottomBar.css"
+import { Link } from "react-router-dom";
```
Comment (shraddha-amber) [inline src/Components/BottomBar/BottomBar.jsx:2]: try to write imports from packages on top and imports from project files below them

File: src/Components/Comments/Comments.jsx [hunk 3]
```diff
@@ -1,32 +1,62 @@
 import React from 'react'
+import { useEffect } from 'react';
```
Comment (shraddha-amber) [inline src/Components/Comments/Comments.jsx]: could've imported both useEffect and useState in same import

File: src/Components/EditProfileModal/EditProfileModal.jsx [hunk 4]
```diff
@@ -14,7 +15,71 @@ const EditProfileModal = ({userdata}) => {
...
+    formData.append("upload_preset", "eahivfql");
+
+    const requestOptions = {
+      method: "POST",
+      body: formData,
+    };
+    fetch(url, requestOptions)
+      .then((res) => res.json())
+      .then((data) => {
+        console.log(data.url);
```
Comment (shraddha-amber) [inline src/Components/EditProfileModal/EditProfileModal.jsx]: remove console logs if not needed

File: src/Components/PostCard/PostCard.jsx [hunk 5]
```diff
@@ -95,39 +112,83 @@ const PostCard = ({postOptions}) => {
...
+          {comments?.length}
+        </div>
         <span className="share material-symbols-rounded ml-8 text-sky-400  rounded-full">
           send
         </span>
-        <span className="bookmark material-symbols-rounded ml-auto mr-8 text-green-400  rounded-full">
+        <span
+          className="bookmark material-symbols-rounded ml-auto mr-8 text-green-400  rounded-full"
+          onClick={() => {
+            bookmarked?.some(
```
Comment (shraddha-amber) [inline src/Components/PostCard/PostCard.jsx:184]: try to separate js logic from jsx, could put this in a function

File: src/Pages/Feed/Feed.jsx [hunk 6]
```diff
@@ -1,12 +1,130 @@
...
+  const { themeObject } = useTheme();
+  const [sortVisible, setSortVisible] = useState(false);
+  const [sort, setSort] = useState("recent");
+  const [homefeedSorted, setHomefeedSorted] = useState(JSON.parse(JSON.stringify(homefeed)));
+  useEffect(() => {
+   sortFeed();
+  },[sort,homefeed])
+  const sortFeed = () => {
+      setHomefeedSorted(
+        homefeed?.slice().sort((a, b) => {
```
Comment (shraddha-amber) [inline src/Pages/Feed/Feed.jsx]: instead of a and b, could give post1 and post2 as variable names

**Review by shraddha-amber — APPROVED** [review]
Ui looks cool
Code is modular
The functionality works properly!

File: src/services/commentsService.js [hunk 7]
```diff
@@ -1,30 +1,83 @@
...
+
+
+export const downvoteComment = async (postId, commentId, authToken) => {
+    try { 
+        const res = await axios.post(`/api/comments/downvote/${postId}/${commentId}`, {
+            headers: {
+                authorization: authToken,
+            }
+        });
+        console.log(res);
```
Comment (Pratik1005) [inline src/services/commentsService.js]: remove log if not needed

**Review by Pratik1005 — APPROVED** [review]
- Functionality working properly
- Great UI
- Good file structure and naming convention

File: src/Pages/Feed/Feed.jsx (hunk 6 above)
Comment (aniketxparihar) [reply src/Pages/Feed/Feed.jsx]: Thank you that makes sense. Changed it.

**Pull request merged**